lldb/source/Plugins/Process/Windows/Common/TargetThreadWindows.cpp

```diff
@@ -175,3 +179,30 @@ Status TargetThreadWindows::DoResume() {
 
   return Status();
 }
+
+const char *TargetThreadWindows::GetName() {
+  Log *log = GetLog(LLDBLog::Thread);
+  HMODULE hModule = ::LoadLibraryW(L"Kernel32.dll");
+  if (hModule) {
+    auto GetThreadDescription =
+        reinterpret_cast<GetThreadDescriptionFunctionPtr>(
+            ::GetProcAddress(hModule, "GetThreadDescription"));
```

Comment:
This code is only compiled on windows, why do we need to get the library handle manually and get the function pointer. Can't we just `#include <processthreadsapi.h>` and then call the function directly?

Reply:
The function was added in Windows 10 1607, which was also the only build (excluding Server 2016) where it isn't available in the header:

> [Windows Server 2016, Windows 10 LTSB 2016 and Windows 10 version 1607: GetThreadDescription is only available by Run Time Dynamic Linking in KernelBase.dll.](https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-getthreaddescription)

Reply:
You are talking about setting thread names, but I implemented getting a thread name from Windows.

Reply:
There is the set name below though.